aten/src/ATen/native/Normalization.cpp

```diff
@@ -365,9 +365,13 @@ std::tuple<Tensor, Tensor, Tensor> batch_norm_backward_cpu_template(
   for (const auto i : c10::irange(2, ndim)) {
     reduce_dims[i - 1] = i;
   }
-
-  auto sum = at::sum(grad_out_, /*dim=*/reduce_dims);
-  auto sum_a = sum.accessor<scalar_t, 1>();
+  // Using float data type for Half sum to avoid overflow
+  // since the representation range of Half is small.
+  auto sum = grad_out_.scalar_type() == kHalf
+      ? at::sum(grad_out_.to(ScalarType::Float), /*dim=*/reduce_dims)
+      : at::sum(grad_out_, /*dim=*/reduce_dims);
+  using sum_t = std::conditional_t<std::is_same_v<scalar_t, at::Half>, float, scalar_t>;
```

Comment:
Also a nit, but would it be useful to use `accscalar_t` from above?

Reply:
Thanks for your comments. Do you mean `using sum_t = std::conditional_t<std::is_same_v<scalar_t, at::Half>, accscalar_t , scalar_t>;`

Reply:
I meant accscalar_t instead of sum_t since for half it should map to float: https://github.com/pytorch/pytorch/blob/af3164039158f38ebe7ff17300c0307ecb0abcd6/aten/src/ATen/AccumulateType.h

Reply:
This will make other types, such as accscalar_t = float for scalar_t = BFloat16 and accscalar_t  = double for float. This PR is currently only intended to fix the half overflow case. Do you mean to make other types also use higher precision for sum?